libs/utils/energy.py

```diff
@@ -275,22 +242,32 @@ def reset(self):
         self._log.debug('RESET: %s', self.channels)
 
         self._aep.start()
-        self.time['start'] = time.time()
 
     def report(self, out_dir, out_energy='energy.json', out_samples='samples.csv'):
-        if self._aep is None:
-            return
+        self._aep.stop()
 
-        # Retrieve energy consumption data
-        csv_file = '{}/{}'.format(out_dir, out_samples)
-        self._sample(csv_file)
+        csv_path = os.path.join(out_dir, out_samples)
+        self._aep.get_data(csv_path)
+        with open(csv_path) as f:
+            # Each column in the CSV will be headed with 'SITE_measure'
+            # (e.g. 'BAT_power'). Convert that to a list of ('SITE', 'mesure)
+            # tuples, then pass that as the `names` parameter to read_csv to get
+            # a nested column index. None of devlib's standard measurement types
+            # have '_' in the name so this use of rsplit should be fine.
+            headers = f.readline().strip().split(',')
```

Comment:
Good idea. Devlib gives us the info we need to figure out the expected column headers, I'll try to add a sanity check.